src/test/java/io/localmotion/picture/controller/PictureControllerIntegrationTest.java

```diff
@@ -0,0 +1,352 @@
+				.build();
+		Picture expectedPicture = picture;
+
+		//When
+		MultipartBody multiPartBody = MultipartBody.builder()
+				.addPart("profilePicture",filenname, picture.getPictureBytes())
+				.build();
+		HttpResponse<String> actualResponse =  client.toBlocking()
+				.exchange(HttpRequest.POST("/api/pictures?type=" + picture.getType() , multiPartBody)
+						.contentType(MediaType.MULTIPART_FORM_DATA)
+						.bearerAuth(this.accessToken), String.class);
+
+		//Then
+		//verify(pictureRepository).uploadPicture(picture);
+		assertEquals(HttpStatus.OK, actualResponse.getStatus());
+		assertTrue(actualResponse.getBody().isPresent());
+	}
+
+	@Test
+	void should_updatePicture_when_forValidFile() {
+		//Given
+		String imageId = UUID.randomUUID().toString();
+		String userId = "test";
+		//String id = UUID.randomUUID().toString();
+		PictureType pictureType = PictureType.PLAYGROUND;
+		final String filename_1 = "image/logo-hartstichting-horizontal.png";
+		final String filename_2 = "image/logo-kwf.png";
+		byte[] pictureBytes = this.getBytesFromFile(filename_1);
+		Picture picture = Picture.builder()
+				.pictureId(PictureId.builder().id(userId).imageId(imageId).build())
+				.type(pictureType)
+				.pictureBytes(pictureBytes)
+				.build();
+		insertPicture(picture);
+		byte[] newImageBytes = this.getBytesFromFile(filename_2);
+		byte[] expectedImageBytes = newImageBytes;
+
+		//When
+		MultipartBody multiPartBody = MultipartBody.builder()
+				.addPart("profilePicture",filename_2, picture.getPictureBytes())
+				.build();
+		HttpResponse<String> actualResponse =  client.toBlocking()
+				.exchange(HttpRequest.PUT("/api/pictures/"+ picture.getPictureId().getImageId() + "?type=" + picture.getType() , multiPartBody)
+						.contentType(MediaType.MULTIPART_FORM_DATA)
+						.bearerAuth(this.accessToken), String.class);
+
+		//Then
+		assertEquals(HttpStatus.OK, actualResponse.getStatus());
+		assertTrue(actualResponse.getBody().isPresent());
+	}
+
+	@Test
+	void should_getPicture_when_forValidImageId() {
+
+		// Given
+		String imageId = UUID.randomUUID().toString();
+		String userId = "test";
+
+		PictureType pictureType = PictureType.PLAYGROUND;
+		String filename = "image/logo-hartstichting-horizontal.png";
+		byte[] pictureBytes = this.getBytesFromFile(filename);
+		Picture uploadPicture = Picture.builder()
+				.pictureId(PictureId.builder().id(userId).imageId(imageId).build())
+				.type(pictureType).pictureMimeType("png")
+				.pictureBytes(pictureBytes)
+				.build();
+		insertPicture(uploadPicture);
+		//pictureRepository.uploadPicture(uploadPicture);
+		//entityManager.persist(uploadPicture);
+
+		// When
+		HttpResponse<byte[]> actualPicture = client.toBlocking()
+				.exchange(HttpRequest.GET("/api/pictures?imageId="+ imageId).bearerAuth(this.accessToken), byte[].class);
+		// Then
+		assertEquals(HttpStatus.OK, actualPicture.getStatus());
+		assertArrayEquals(uploadPicture.getPictureBytes(), actualPicture.getBody().get());
+	}
+
```

Comment:
Include test cases for
- when the file size is too large
- when the picture is too big, that is width > maximum or height > maximum